### DISCUSSION TITLE
Isolated node for testing

### DISCUSSION THREAD

**Discussion by itamarst** [PurpleI2P/i2pd#1929] [Q&A]
Hi,

Not an i2p user, really, but I'm working on software (Tahoe-LAFS) that integrates with it. I would like to write an end-to-end test; in theory it has one but I think it was actually broken. Since the goal is to run a test, I don't want to connect to the global i2p network, I just want to run locally in isolation.

1. I run i2pd with `--reseed.urls http://localhost:1/`, which seems to make it run in isolation.
2. Next, I want to listen in isolation and then connect to that. What options should I be using to do so? `inbounds.hops:0,outbound.hops:0` doesn't seem sufficient, it's complaining at some point about "Destination: Can't publish LeaseSet, no more floodfills found" and then later "Tunnels: Can't select next hop for 0mYfFQSuvemv0Kii1ZQmYonN046ju2taevnbBszhZQ8= Tunnels: Can't create outbound tunnel, no peers available".

Anything else I should be doing? Is this even possible?

**Comment by anikey-from-i2p** [comment]
After a quick web search i have arrived at the following link: https://github.com/h-phil/i2pd-testnet-kubernetes

It has some other options/links listed at the end.

Of course if this testnet thing does not work for you, you can just keep an external i2pd node (connected to the main network) and connect to that via SAM (if you're using that) multiple times with different ids, it should in theory work like that. In that case you should reduce the amount of hops to get more reliability and to spend less resources.
